content/doc/developer/security/security-architecture.adoc

```diff
@@ -0,0 +1,190 @@
+----
+The permission is always checked against the "it" object, so that needs to be an `AccessControlled` object.
+
+NOTE: This only prevents access to the view (e.g. configuration form), and does not prevent submissions to the form submission endpoints.
+This will still need to be done as described in the previous section.
+
+==== Modify page rendering based on permissions
+
+==== Disabling a part of page rendering if the user doesn't have a permission
+Page rendering can be modified based on the user's permissions.
+For example, if the user cannot delete a project, it does not make sense to show a link to do that.
+To do this, write Jelly like this:
+----
+<j:if test="${h.hasPermission(app.ADMINISTER)}">
+  ...
+</j:if>
+----
+
+NOTE: This is not to be confused with the `checkPermission` invocation in your operation.
+Users can still hit the URL directly, so you still need to protect the operation itself, in addition to disabling the UI rendering
+
+=== Authentication ways
+
+Jenkins uses
+link:https://spring.io/projects/spring-security[Spring Security] as its security engine.
+With no special plugins that manage authentication installed, a Jenkins instance is packaged
+with the following authentication ways:
+
+* Web UI
+** When you use the form on the login page, using the fields `j_username` and `j_password`
+* REST API
+** Using Basic with login / *password*
+** Using Basic with login / *apiToken*
+* Jenkins CLI jar
+** (deprecated) using remoting transport with login / logout command
+** (deprecated) username / password as parameters on each command
+** `-auth` argument with username:password or username:apiToken that will do something like HTTP calls
+** using SSH transport mode with your local keys
+* CLI over SSH
+** directly using the native SSH command, without Jenkins CLI
+
+=== Authentication flow
+
+the processing flow differs drastically depending on the authentication method you use.
+By flow we mean the involved classes that check your credentials for validity.
+
+==== Web UI and REST API
+
+image:web_rest_flow.svg["Web UI and REST API flow", role=center]
+
+In the diagram above, each arrow indicates a way to authenticate.
+
+Both the Web UI and the REST API using login / password will flow in the same `AbstractPasswordBasedSecurityRealm`
+that delegates the real check to the configured `SecurityRealm`.
+The credentials are retrieved for the first method by retrieving information in the POST and for the second by using the Basic Authentication (in header).
+A point that is important to mention here, the Web UI is the only way (not deprecated) that use the Session to save the credentials.
+
+For the login / apiToken calls, the `BasicHeaderApiTokenAuthenticator` manages to check if the apiToken corresponds to the user with the given login.
+
+==== CLI (SSH and native)
+
+For the CLI part, the things become a bit more complicated, not by the complexity but more by the multiplicity of way to connect.
```

Comment:
```suggestion
For the CLI part, the things become a bit more complicated, not by the complexity but more by the multiplicity of ways to connect.
```

Reply:
This sentence needed a few more tweaks so I did a separate commit and am marking this as resolved.